op-node/p2p/gossip.go

```diff
@@ -142,12 +145,19 @@ func BuildGlobalGossipParams(cfg *rollup.Config) pubsub.GossipSubParams {
 
 // NewGossipSub configures a new pubsub instance with the specified parameters.
 // PubSub uses a GossipSubRouter as it's router under the hood.
-func NewGossipSub(p2pCtx context.Context, h host.Host, cfg *rollup.Config, gossipConf GossipSetupConfigurables, m GossipMetricer) (*pubsub.PubSub, error) {
+func NewGossipSub(p2pCtx context.Context, h host.Host, g ConnectionGater, cfg *rollup.Config, gossipConf GossipSetupConfigurables, m GossipMetricer, log log.Logger) (*pubsub.PubSub, error) {
 	denyList, err := pubsub.NewTimeCachedBlacklist(30 * time.Second)
 	if err != nil {
 		return nil, err
 	}
 	params := BuildGlobalGossipParams(cfg)
+	// TODO: make this configurable behind a cli flag - disabled or default
+	peerScoreParams, err := GetPeerScoreParams("default", cfg)
```

Comment:
Feel free to ignore this, but if peer scoring is disabled, it may be worth not even creating any of the scoring infrastructure. ie when disabled, don't create any score params or thresholds, don't create the scorer and don't add the `pubsub.WithPeerScore` and `pubsub.WithPeerScoreInspect` options at all.

That way disabling peer scoring acts as a feature toggle and if some problem is encountered with peer scoring, it can be completely disabled. Whereas if we just disable it by using a different set of params, any bugs in the scorer would still cause issues and potentially the "disabled" params may wind up not as disabled as we intended etc. Also saves us from needing to test those disabled params.

Reply:
Great suggestion, I think for now I'll leave it to be configurable so we can experiment in a devnet environment to gain better intuition for the parameters we need.